pybundlr/pybundlr.py

```diff
@@ -1,231 +1,159 @@
-            args[i+1] = args[i+1][2:]
-            break
-    return " ".join(args)
-
-def _safe_print(cmd: str) -> str:
-    """Avoid dangerous logging. E.g. don't show whole private key"""
-    args = cmd.split(" ")
-    for i, arg in enumerate(args):
-        if arg == "-w":
-            args[i+1] = args[i+1][:6] + "..." #truncate private key
-    return " ".join(args)
-    
-
-#==========================================================================
-#eth convenience functions
-
-def eth_address(eth_private_key:str) -> str:
-    account = web3.eth.Account.from_key(eth_private_key)
-    return account.address
 
+@enforce_types
+def get_price_for_uploading_this_file(file_name: str, currency_name: str, private_key: str) -> float:
+    r"""
+    Returns the price in decimal units that will cost to upload this file
 
-def bal_on_ethereum(eth_address:str) -> int:
-    """
-    Returns ether balance on Ethereum mainnet
-    
     Parameters:
-      eth_address - address on eth mainnet
+      file - a string with the fully qualified filename
+      currency_name - name of one of Bundlr's supported currencies
+      private_key - user's private key
 
     Returns:
-      bal - ether balance, denominated in wei
+      string - amount in atomic units that would cost to upload this number of bytes at this moment
     """
-    return _w3_ethereum().eth.get_balance(eth_address)
+    bundlr = init(currency_name, private_key)
+    size = bundlr.createTransaction(file_name).size
+    total_cost_atomic = bundlr.getPrice(size)
+    total_cost = eval_js(''' bundlr.utils.unitConverter(total_cost_atomic).valueOf() ''')
+    return total_cost
 
 
-def bal_on_matic(eth_address:str) -> int:
-    """
-    Returns ether balance on Matic
-    
+@enforce_types
+def upload_data(file_name: str, currency_name: str, private_key: str):
+    r"""
+    Uploads data to Arweave
+
     Parameters:
-      eth_address - address on eth mainnet
+      file - a string with the fully qualified filename
+      currency_name - name of one of Bundlr's supported currencies
+      private_key - user's private key
 
     Returns:
-      bal - ether balance, denominated in wei
-    """
-    return _w3_matic().eth.get_balance(eth_address)
-
-def _w3_ethereum():
-    """Returns - web3 object pointed to Ethereum mainnet"""
-    return Web3(Web3.HTTPProvider(ETH_NODE_URL))
-
-def _w3_matic():
-    """Returns - web3 object pointed to Matic (Polygon) mainnet """
-    return Web3(Web3.HTTPProvider(MATIC_NODE_URL)) 
-
-
+      string|Exception - a string with the success message or an exception explaining the error
+    """
+    balance = my_balance(currency_name=currency_name, private_key=private_key)
+    price_to_upload = get_price_for_uploading_this_file(file_name=file_name, currency_name=currency_name, private_key=private_key)
+
+    if balance > price_to_upload:
+        try:
+            bundlr = init(currency_name, private_key)
+            tx = bundlr.createTransaction(file_name)
```

Comment:
Hi @theref! Thanks very much for your interest in pybundlr. Sorry for the delay (I was sick). 

FYI maintaining & improving pybundlr is not a priority for Ocean right now: we deprecated our need for it in Feb 2023. 

That's why this PR has been hanging. We kept the PR around in case we found ourselves coming back to pybundlr wanting to improve it.

So if you'd like to improve pybundlr, by all means go for it! Perhaps the best way is: simply fork the repo and then run with it how you like. Therefore we're not a bottleneck to  you.

We'd be very interested to see what you do with it:)

Reply:
Sorry to hear that you've been sick, but thanks for replying anyway! :)

Out of interest, did you move away from bundlr? or to a new toolchain? just trying to figure out all my options